OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc

```diff
@@ -647,8 +647,10 @@ fmi2Status fmi2ExitInitializationMode(fmi2Component c)
     }
   }
 
-  /* TODO: Simulation stop time is needed to calculate the sample events beforehand. */
-  initSample(comp->fmuData, comp->threadData, comp->fmuData->localData[0]->timeValue, 100 /*should be stopTime*/);
+  /* use defined stopTime, if stopTimeDefined is given to calculate the sample events beforehand.
+   * TODO: when stopTime is not defined we use an arbitrary constant 100.0, maybe issue a warning
+   */
+  initSample(comp->fmuData, comp->threadData, comp->fmuData->localData[0]->timeValue, comp->stopTimeDefined ? comp->stopTime : 100.0 /* default stopTime */);
```

Comment:
@adrpo Can we recalculate the events if we exceed `100.0`?

Reply:
I wondered about that, I don't really know. It was 100 before which seems arbitrary.

Reply:
I know that we had the same odd behavior before. However, it is a major issue for certain relevant applications and people probably don't know about it. And I would rather like to get this fixed than to tell people about our arbitrary 100.

Reply:
Agreed, but somebody else needs to fix it as I'm not sure how. I guess one should monitor if it goes about the default value and somehow re-allocate the data in initSample.